quinn-proto/src/connection.rs

```diff
@@ -790,24 +776,41 @@ impl Connection {
             }
             Err(None) => {
                 debug!(self.log, "failed to authenticate packet");
-                return;
+                if self.check_for_stateless_reset(mux, &packet) {
+                    Err(ConnectionError::Reset)
+                } else {
+                    return;
+                }
             }
             Ok(number) => {
-                if let Some(number) = number {
+                let duplicate = if let Some(number) = number {
                     if self.dedup.insert(number) {
-                        warn!(
-                            self.log,
-                            "discarding possible duplicate packet {packet}",
-                            packet = number
-                        );
+                        Some(number)
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                };
+
+                if let Some(number) = duplicate {
+                    warn!(
+                        self.log,
+                        "discarding possible duplicate packet {packet}",
+                        packet = number
+                    );
```

Comment:
Let's move this warning into the `else` case just below, since honoring a connection reset does not constitute discarding the packet.

Reply:
Done.